pkgs/_analyzer_macros/lib/analyzer_macro_implementation.dart

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:_macro_host/macro_host.dart';
+// ignore: implementation_imports
+import 'package:analyzer/src/summary2/macro_injected_impl.dart' as injected;
+import 'package:dart_model/dart_model.dart';
+import 'package:macro_service/macro_service.dart';
+import 'package:macros/macros.dart';
+// ignore: implementation_imports
+import 'package:macros/src/executor.dart' as injected;
+
+/// Injected macro implementation for the analyzer.
+class MacroImplementation implements injected.MacroImplementation, HostService {
+  final Uri packageConfig;
+  final Map<String, String> macroImplByName;
+  late final MacroHost _host;
+
+  MacroImplementation._(this.packageConfig, this.macroImplByName);
+
+  /// Starts the macro host.
+  ///
+  /// [packageConfig] is the package config of the workspace of the code being
+  /// edited.
+  ///
+  /// [macroImplByName] identifies macros, it's a placeholder until we identify
+  /// macros using package config. Keys are macro annotation qualified names
+  /// (`uri#name`) and values are macro implementation qualified names.
+  static Future<MacroImplementation> start({
+    required Uri packageConfig,
+    required Map<String, String> macroImplByName,
+  }) async {
+    final result = MacroImplementation._(packageConfig, macroImplByName);
+    result._host = await MacroHost.serve(
+        macroImplByName: macroImplByName, service: result);
+    return result;
+  }
+
+  @override
+  injected.MacroPackageConfigs get packageConfigs => MacroPackageConfigs(this);
+
+  @override
+  injected.MacroRunner get runner => MacroRunner(this);
+
+  @override
+  Future<Response?> handle(MacroRequest request) async {
+    if (request.type != MacroRequestType.queryRequest) return null;
+    return Response.queryResponse(QueryResponse());
+  }
+}
+
+class MacroPackageConfigs implements injected.MacroPackageConfigs {
+  final MacroImplementation impl;
+
+  MacroPackageConfigs(this.impl);
+
+  @override
+  bool isMacro(Uri uri, String name) =>
```

Comment:
Seems a bit weird that this is just a wrapper which calls a function by the same name, but which takes slightly different args. Can we unify the signatures? Similar patterns below

Reply:
That's because the injected API assumes the package config URI is known from context, but the `_macro_host` API wants it passed in.

We could update `_macro_host` API to match, also assuming it from context, but it's not clear to me yet what the scope of one `MacroHost` will be, it only makes sense if it's going to be for a single package config.

So, maybe :)